packages/istanbul-reports/lib/clover/index.js

```diff
@@ -135,13 +135,35 @@ class CloverReport extends ReportBase {
                 count,
                 type: 'stmt'
             };
-            const branchDetail = branchByLine[k];
+            const branchDetail = branchDetails[k];
 
-            if (branchDetail) {
-                attrs.type = 'cond';
-                attrs.truecount = branchDetail.covered;
-                attrs.falsecount = branchDetail.total - branchDetail.covered;
+            if (!branchDetail || branchDetail.type === 'switch') {
```

Comment:
What is the effect of adding a `line` tag here and returning immediately, will this result in 100% line coverage and no missing branch coverage?

Reply:
It is from the [original code](https://github.com/istanbuljs/istanbuljs/blob/fb8cb4bb99c9e5d36c8b011338d2ab5d3c72ff9c/packages/istanbul-reports/lib/clover/index.js#L140-L145):

```javascript
if (branchDetail) {
    attrs.type = 'cond';
    attrs.truecount = branchDetail.covered;
    attrs.falsecount = branchDetail.total - branchDetail.covered;
}
this.xml.inlineTag('line', attrs);
```

but the true case is more complex so I move the false case into the brackets and change the condition to `!branchDetail`.

As to `branchDetail.type === 'switch'`. Thanks to you, I found a situation as you said: 

```javascript
switch (num) {
    case 0:
    case 1:
        console.log('Hello');
        break;
}
```

When we use num=1 to test this switch, all lines are covered but branch not.

Maybe we should treat `switch` the same as `binary-expr`.